tools/osv-scanner/cmd/osv-scanner/config.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"golang.org/x/exp/slices"
+)
+
+type ConfigManager struct {
+	// Override to replace all other configs
+	overrideConfig *Config
+	// Config to use if no config file is found alongside manifests
+	defaultConfig Config
+	// Cache to store loaded configs
+	configMap map[string]Config
+}
+
+type Config struct {
+	IgnoredVulns []IgnoreLine
+	LoadPath     string
+}
+
+type IgnoreLine struct {
+	ID          string    `toml:"id"`
+	IgnoreUntil time.Time `toml:"ignoreUntil"`
+	Reason      string    `toml:"reason"`
+}
+
+func (c *Config) ShouldIgnore(vulnID string) (bool, IgnoreLine) {
+	index := slices.IndexFunc(c.IgnoredVulns, func(elem IgnoreLine) bool { return elem.ID == vulnID })
+	if index == -1 {
+		return false, IgnoreLine{}
+	}
+	ignoredLine := c.IgnoredVulns[index]
+	if ignoredLine.IgnoreUntil.IsZero() {
+		// If IgnoreUntil is not set, should ignore.
+		return true, ignoredLine
+	}
+	// Should ignore if IgnoreUntil is still after current time
+	return ignoredLine.IgnoreUntil.After(time.Now().UTC()), ignoredLine
```

Comment:
`IgnoreUntil` also gets parsed in as UTC right? what happens when someone provides a non-UTC timestamp?

Reply:
IgnoreUntil should be an ISO 8601 datestring, which can contain an offset for non-UTC timestamp. If they don't pass in an offset I think this will assume it is UTC, though I haven't tested this.

I don't think this is a big concern since at most it's off by 12 hours, but the expiry is designed to be used with a granularity of days, not hours.

Reply:
If the objective is to be purely date-based, maybe make this more unambiguous by only taking a date. https://toml.io/en/v1.0.0#local-date seems to support this.

Reply:
We should still get it right because it can be confusing to people. 

I wonder if  just calling `IgnoreUntil.UTC()` is sufficient here.

Reply:
+1 to just limiting this to a date as @andrewpollock suggested, but there's still the UTC question to resolve.

Reply:
Just tested the time UTC/not UTC issue. The `.After` comparison takes into account the timezone specified, so calling `.UTC()` in either part of the comparison doesn't change the result. 

If no timezone is specified, it assumes that it is **local** time, not UTC time as I guessed above. 

> If the objective is to be purely date-based, maybe make this more unambiguous by only taking a date. https://toml.io/en/v1.0.0#local-date seems to support this.

I don't see any way to restrict the type of date specified for the toml parser without using a completely custom type. I can change the examples in the readme and the osv-scanner files we use to use local dates instead though.

Reply:
Sounds good, thanks!